extras/docker/fromsource/heketi-start.sh

```diff
@@ -74,4 +85,32 @@ if [[ -d "${BACKUPDB_PATH}" ]]; then
     fi
 fi
 
-/usr/bin/heketi --config=/etc/heketi/heketi.json
+# if NEED_TOPOLOGY_LOAD is not empty, start the service in the background, and
+# load the topology.json from the path in HEKETI_TOPOLOGY_FILE
+if [ -z "${NEED_TOPOLOGY_LOAD}" ]; then
+    # by default just start in the foreground
+    /usr/bin/heketi --config=/etc/heketi/heketi.json
+else
+    # start hketi in the background
+    /usr/bin/heketi --config=/etc/heketi/heketi.json &
+
+    # wait until heketi replies
+    while ! curl http://localhost:8080/hello; do
+        sleep 0.5
+    done
+
+    # load the topology
+    if [ -n "${HEKETI_ADMIN_KEY}" ]; then
+        HEKETI_SECRET_ARG="--secret='${HEKETI_ADMIN_KEY}'"
+    fi
+    heketi-cli --user=admin "${HEKETI_SECRET_ARG}" topology load --json="${HEKETI_TOPOLOGY_FILE}"
+    if [ $? -ne 0 ]; then
+        # something failed, need to exit with an error
+        kill %1
+	echo "failed to load topology from ${HEKETI_TOPOLOGY_FILE}"
+	exit 1
```

Comment:
These lines appear to be mixing spaces to indent with tabs. We should be consistent with the rest of the file (spaces).

Reply:
Oh, of course!

Reply:
Done!